frontend/lib/shared/animations/invalid_number_pad_input_animation.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:flutter/material.dart';
+
+class InvalidNumberPadInputAnimation extends StatefulWidget {
+  final String textValue;
+  final bool shouldAnimate;
+  final TextStyle? textStyle;
+
+  const InvalidNumberPadInputAnimation(
+      {required this.textValue,
+      required this.shouldAnimate,
+      this.textStyle,
+      super.key});
+
+  @override
+  State<InvalidNumberPadInputAnimation> createState() =>
+      _InvalidNumberPadInputAnimationState();
+}
+
+class _InvalidNumberPadInputAnimationState
+    extends State<InvalidNumberPadInputAnimation>
+    with SingleTickerProviderStateMixin {
+  late final AnimationController _controller = AnimationController(
+    duration: const Duration(milliseconds: 375),
+    vsync: this,
+  );
+  late final Animation<Offset> _offsetAnimation = Tween<Offset>(
+          begin: const Offset(0.0, 0.0),
+          end: Offset(0.75 / widget.textValue.length, 0.0))
+      .animate(CurvedAnimation(
+    parent: _controller,
+    curve: Curves.elasticIn,
```

Comment:
will defer to @wesbillman on which `Curve` is most appropriate here

Reply:
I think any curve that looks ok is good for now. We could spend decades on animation polish :)